share/wake/lib/core/integer.wake

```diff
@@ -13,78 +13,80 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 package wake
 
 # The Integer type has unbounded precision (aka a "big" integer).
 # Any wake literal like 12331_1232 or 0x3123 is an Integer.
 
+
 # Unary positive sign operator for Integer values.
 # (+5) = 5
-export def +(x: Integer): Integer =
-    x
+export def +(x: Integer): Integer = x
+
 
 # Unary negative sign operator for Integer values.
 # (-5) = 0-5
-export def -(x: Integer): Integer =
-    (\_ prim "neg") x
+export def -(x: Integer): Integer = (\_ prim "neg") x
+
 
 # Unary two's complement operator for Integer values.
 # ~0 = -1
 # ~4 = -5
-export def ~(x: Integer): Integer =
-    (\_ prim "com") x
+export def ~(x: Integer): Integer = (\_ prim "com") x
+
 
 # Binary addition operator for Integer values.
 # 1 + 2 = 3
 # 1 + 5 = 6
-export def (x: Integer) + (y: Integer): Integer =
-    (\_\_ prim "add") x y
+export def (x: Integer) + (y: Integer): Integer = (\_\_ prim "add") x y
+
 
 # Binary subtraction operator for Integer values.
 # 2 - 1 =  1
 # 3 - 4 = -1
-export def (x: Integer) - (y: Integer): Integer =
-    (\_\_ prim "sub") x y
+export def (x: Integer) - (y: Integer): Integer = (\_\_ prim "sub") x y
+
 
 # Binary multiplication operator for Integer values.
 #  3 *   4  = 12
 # -3 * (-4) = 12
-export def (x: Integer) * (y: Integer): Integer =
-    (\_\_ prim "mul") x y
+export def (x: Integer) * (y: Integer): Integer = (\_\_ prim "mul") x y
+
 
 # Binary division operator for Integer values.
 # 12 / 3 =  4
 # 13 / 3 =  4
 # -8 / 4 = -2
-export def (x: Integer) / (y: Integer): Integer =
-    (\_\_ prim "div") x y
+export def (x: Integer) / (y: Integer): Integer = (\_\_ prim "div") x y
+
 
 # Binary remainder operator for Integer values.
 # 11 % 5 = 1
 #  4 % 5 = 5
 #  7 % 5 = 2
-export def (x: Integer) % (y: Integer): Integer =
-    (\_\_ prim "mod") x y
+export def (x: Integer) % (y: Integer): Integer = (\_\_ prim "mod") x y
```

Comment:
I think I'd prefer `(\_\_ prim "mod") x y` to be formatted as `(\_ \_ prim "mod") x y`, with a space between the lambdas. I think that's particularly so for actual named arguments, like `\x \y (x + y)` (not sure if the formatter already picks that up and puts a space in between), and to maintain consistency between the cases.

Reply:
There just isn't a lambda formatting rule right now. Taking suggestions. @ag-eitilt brought this up in another thread as well

Reply:
Ah, gotcha. I'll look for Sam's comment on this, but I'm wondering now if there should be a delineation between the two. This is because even when I was writing my comment, I was wondering if no space between the lambda arguments is fine for `\_\_` but then having the space is preferred for `\x \y`. I went with my comment because I sided on spaces for `\x \y` and consistency.

Some of the formatting rules for lambdas may depend on the result of https://github.com/sifive/wake/pull/1044.